tests/Owin.Security.OpenIdConnect.Server.Tests/OAuth2AuthorizationClientCredentialsGrantTests.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.Owin.Testing;
+using Xunit;
+
+namespace Owin.Security.OpenIdConnect.Server.Tests {
+    public class OAuth2AuthorizationClientCredentialsGrantTests {
+        [Fact]
+        public async Task ClientCredentials_MissingCredentials_BadRequest() {
+            var server = TestServer.Create(app => { app.UseOpenIdConnectServer(new OpenIdConnectServerOptions { AllowInsecureHttp = true }); });
+
+            var openIdConnectMessage = new OpenIdConnectMessage {
+                RequestType = OpenIdConnectRequestType.TokenRequest,
```

Comment:
Ah yeah, I didn't realize that you added such a method.
In my defense, you haven't used it here :smile: